### DISCUSSION TITLE
UVtools silently exits with 255 and no clue what's the problem

### DISCUSSION THREAD

**Discussion by WizardUli** [sn4k3/UVtools#656] [General]
Hello. I decided to package UVtools for Nix (as a Nix flake) without using any pre-built binaries or Appimages.
So far I'm just trying to build and run it manually and I can successfully `dotnet run --project UVtools.Cmd` which works as expected but `dotnet run --project UVtools.WPF` returns 255 without any output to stdout or stderr.

It does build successfully (`dotnet build` or `dotnet run -v normal` shows that building succeeds) but running the binary does not open any window or produce anything except of a return value of 255. Running it without `dotnet run` i.e. directly from the `bin/....` dir produces the same result.

When run with `--cmd xyz` it tries to use xdg-open and delegate the command to `UVtoolsCmd` so at least some part of the `UVtools` binary really runs. How can I diagnose it further? Is there some magical switch to turn on logging / debug output? Thanks.

**Comment by sn4k3** [comment]
You need to use the [createRelease.sh](https://github.com/sn4k3/UVtools/blob/master/build/createRelease.sh) script which copies the required dependency to run (OpenCV). If you want to do it manually extract [this zip](https://github.com/sn4k3/UVtools/raw/master/build/platforms/linux-x64/libcvextern.zip) to build folder

**Reply by WizardUli** [reply]
No. I explicitly said I do not want to use any pre-built binaries which of course includes libraries. I can build and package _emgucv_ myself.

And anyway, the problem is that inside `Program.cs` there is a method `HandleUnhandledException` which does not print anything from the `Exception ex` to stderr nor to stdout and just exits with `-1`. Something like this is completely unheard of in Linux/Mac/BSD world (unless very explicitly requested by `--quiet` parameter).

**Reply by sn4k3** [reply]
Without `libcvextern.so` it will not run, you must have that in same directory.
UI apps often dont send console messages.
